Spam error on console with task exception

Task # for Mydog generated an exception java.lang.nullpointerExepctionnull
[https://imgur.com/lqChI0D](url)

[29.05 18:40:15] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:15] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:18] [Server] [INFO] Nightcodex issued server command: /lp reloadconfig
[29.05 18:40:19] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:19] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:21] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:21] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:23] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:23] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:24] [Server] [WARN] [MyDog] Task #15 for MyDog v0.1.4 generated an exception
[29.05 18:40:24] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:24] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:24] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:26] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:26] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:28] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:28] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:29] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:29] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:32] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:32] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:34] [Server] [WARN] [MyDog] Task #15 for MyDog v0.1.4 generated an exception
[29.05 18:40:34] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:34] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:34] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:36] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:36] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:37] [Server] [INFO] [UltimateAutoRestart] » Restarting in 30s!
[29.05 18:40:37] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:37] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:39] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:39] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:41] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:41] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:43] [Server] [WARN] [MyDog] Task #15 for MyDog v0.1.4 generated an exception
[29.05 18:40:43] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:43] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:43] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:44] [Server] [INFO] [ClearLag] 648 Mobs have been removed!
[29.05 18:40:44] [Server] [INFO] [ClearLag] You just removed 37 entities!
[29.05 18:40:44] [Server] [INFO] [ClearLag] Server activity has been halted!
[29.05 18:40:45] [Server] [INFO] [LP] LOG > (Nightcodex) [G] (default)
[29.05 18:40:45] [Server] [INFO] [LP] LOG > webeditor add mydog.teleport true
[29.05 18:40:45] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:45] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:47] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:47] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:49] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:49] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:51] [Server] [WARN] [MyDog] Task #16 for MyDog v0.1.4 generated an exception
[29.05 18:40:51] [Server] java.lang.NullPointerExceptionnull
[29.05 18:40:53] [Server] [INFO] Kuga_ issued server command: /mydog dogs
[29.05 18:40:53] [Server] [ERROR] null
[29.05 18:40:53] [Server] org.bukkit.command.CommandExceptionUnhandled exception executing command 'mydog' in plugin MyDog v0.1.4
[29.05 18:40:53] [Server] 	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
[29.05 18:40:53] [Server] 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
[29.05 18:40:53] [Server] 	at org.bukkit.craftbukkit.v1_19_R3.CraftServer.dispatchCommand(CraftServer.java:907) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at org.bukkit.craftbukkit.v1_19_R3.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264) ~[paper-1.19.4.jar:?]
[29.05 18:40:53] [Server] 	at net.minecraft.commands.Commands.performCommand(Commands.java:322) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.commands.Commands.performCommand(Commands.java:306) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.server.network.ServerGamePacketListenerImpl.performChatCommand(ServerGamePacketListenerImpl.java:2297) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$20(ServerGamePacketListenerImpl.java:2257) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:59) ~[?:?]
[29.05 18:40:53] [Server] 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1342) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1319) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1312) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:114) ~[?:?]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1428) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1174) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.19.4.jar:git-Paper-540]
[29.05 18:40:53] [Server] 	at java.lang.Thread.run(Thread.java:833) ~[?:?]

--- a/src/main/java/dk/fido2603/mydog/objects/Dog.java
+++ b/src/main/java/dk/fido2603/mydog/objects/Dog.java
@@ -5,6 +5,7 @@ import dk.fido2603.mydog.utils.ColorUtils;
 import org.bukkit.*;
 import org.bukkit.attribute.Attribute;
 import org.bukkit.attribute.AttributeInstance;
+import org.bukkit.entity.Entity;
 import org.bukkit.entity.Player;
 import org.bukkit.entity.Wolf;
 
@@ -303,15 +304,22 @@ public class Dog {
     }
 
     public Location getDogLocation() {
-        if (MyDog.instance().getServer().getEntity(dogId) == null || !MyDog.instance().getServer().getEntity(dogId).isValid()) {
+        Entity dogEntity = MyDog.instance().getServer().getEntity(dogId);
+        if (dogEntity == null || !dogEntity.isValid()) {
             if (MyDog.getDogManager().getDogsConfig().getString(dogId.toString() + ".LastSeen.World") != null) {
-                return new Location(MyDog.instance().getServer().getWorld(MyDog.getDogManager().getDogsConfig().getString(dogId.toString() + ".LastSeen.World")), MyDog.getDogManager().getDogsConfig().getInt(dogId.toString() + ".LastSeen.X"), MyDog.getDogManager().getDogsConfig().getInt(dogId.toString() + ".LastSeen.Y"), MyDog.getDogManager().getDogsConfig().getInt(dogId.toString() + ".LastSeen.Z"));
+                String lastSeenWorld = MyDog.getDogManager().getDogsConfig().getString(dogId.toString() + ".LastSeen.World");
+                if (lastSeenWorld == null) {
+                    return null;
+                }
+                return new Location(MyDog.instance().getServer().getWorld(lastSeenWorld), MyDog.getDogManager().getDogsConfig().getInt(dogId.toString() + ".LastSeen.X"), MyDog.getDogManager().getDogsConfig().getInt(dogId.toString() + ".LastSeen.Y"), MyDog.getDogManager().getDogsConfig().getInt(dogId.toString() + ".LastSeen.Z"));
             }
             return null;
         }
 
-        this.location = MyDog.instance().getServer().getEntity(dogId).getLocation();
-        MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.World", location.getWorld().getName());
+        this.location = dogEntity.getLocation();
+        if (this.location.getWorld() != null) {
+            MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.World", location.getWorld().getName());
+        }
         MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.X", location.getX());
         MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.Y", location.getY());
         MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.Z", location.getZ());
@@ -321,12 +329,15 @@ public class Dog {
     }
 
     public boolean saveDogLocation() {
-        if (MyDog.instance().getServer().getEntity(dogId) == null || !MyDog.instance().getServer().getEntity(dogId).isValid()) {
+        Entity dogEntity = MyDog.instance().getServer().getEntity(dogId);
+        if (dogEntity == null || !dogEntity.isValid()) {
             return false;
         }
 
-        this.location = MyDog.instance().getServer().getEntity(dogId).getLocation();
-        MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.World", location.getWorld().getName());
+        this.location = dogEntity.getLocation();
+        if (this.location.getWorld() != null) {
+            MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.World", location.getWorld().getName());
+        }
         MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.X", location.getX());
         MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.Y", location.getY());
         MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.Z", location.getZ());
@@ -635,14 +646,7 @@ public class Dog {
         }
 
         // Save the Dog's last seen location
-        if (this.location == null) {
-            this.location = getDogLocation();
-        } else {
-            MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.World", location.getWorld().getName());
-            MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.X", location.getX());
-            MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.Y", location.getY());
-            MyDog.getDogManager().getDogsConfig().set(dogId.toString() + ".LastSeen.Z", location.getZ());
-        }
+        this.location = getDogLocation();
 
         // Give the Dog a level
         if (MyDog.instance().useLevels) {

